roofit/roofitcore/src/RooCintUtils.cxx

```diff
@@ -122,21 +123,15 @@ namespace RooCintUtils
 
     ClassInfo_t* cls = gInterpreter->ClassInfo_Factory(className);
     DataMemberInfo_t* dm = gInterpreter->DataMemberInfo_Factory(cls);
-    while (gInterpreter->DataMemberInfo_Next(dm)) {
-      // Check if this data member represents an enum value
-      // there is no "const" with CLING 
-      //if (string(Form("const %s",typeName))==gInterpreter->DataMemberInfo_TypeName(dm)) {  // this for 5.34
-      if (string(Form("%s",typeName))==gInterpreter->DataMemberInfo_TypeName(dm)) {
-	if (string(value)==gInterpreter->DataMemberInfo_Name(dm)) {
-          gInterpreter->DataMemberInfo_Delete(dm);
-          gInterpreter->ClassInfo_Delete(cls);
-	  return kTRUE ;
-	}
-      }
-    }
-    gInterpreter->DataMemberInfo_Delete(dm);
-    gInterpreter->ClassInfo_Delete(cls);
-    return kFALSE ;
+
+    Bool_t res;
+
+    if (string(value)==gInterpreter->DataMemberInfo_Name(dm))
```

Comment:
What about just `return gInterpreter->DataMemberInfo_Name(dm) == string(value);`.

Reply:
Sounds good

Reply:
Is it Ok if we return bool not Bool_t?

Reply:
> Is it Ok if we return bool not Bool_t?

They are the same type :)

Reply:
> What about just return gInterpreter->DataMemberInfo_Name(dm) == string(value);.

Actually no ... at least not that simply as the code is missing:
```
  gInterpreter->DataMemberInfo_Delete(dm);
  gInterpreter->ClassInfo_Delete(cls);
```
which must be called before returning.

Reply:
Maybe in general we should introduce RAII helpers for these.

Reply:
This is a good idea.  We can even leverage the 'only-forward-declared-but-distinct' types (See TDictionary.h) to help out.   All we need is a way to associate the type with the factory function (i.e. ClassInfo_t to ClassInfo_Factory and ClassInfo_Delete) .... maybe we can use unique_ptr with a custom deleter as the RAII ..>)